### PR TITLE
Improve typescrypt definitions

### PR DESCRIPTION
**Pull request by anxolin**
Includes:
- [X] Typechain is super outdated, so I had to fix the definitions, I think this `src/contract` dir can be improved overtime and can be moved to its own package 
- [X] Now the watchEvents is returning a much more specific type, what makes it more convenient for formatting the message
- [X] Renamed the `src/contracts/types.d.ts` to `types.ts` cause the lintern doesn't validate it otherwise, so it's useful to get the errors. 
- [X] Rename `src/contracts/StablecoinConverter.d.ts` to `StablecoinConverter.ts` for the same reason (not done yet cause I have a small issue with the constructor)

Additionally:
* I created in another issue some problems with typescrypt https://github.com/gnosis/dex-telegram/issues/14
* Applied some solutions for typescript issues in: #26

### CODE REVIEW

File: src/repos/dfusion/DfusionRepoImpl.ts [hunk 1]
```diff
@@ -26,14 +26,14 @@ export class DfusionRepoImpl implements DfusionRepo {
   public watchOrderPlacement (params: WatchOrderPlacementParams) {
     this._contract.events
       .OrderPlacement({}, error => params.onError(error))
-      .on('connected', () => {
-        log.info('Starting to listen for new orders')
+      .on('connected', subscriptionId => {
+        log.debug('Starting to listen for new orders. SubscriptionId: %s', subscriptionId)
```
Comment (alfetopito) [inline src/repos/dfusion/DfusionRepoImpl.ts]: question: any special reason for `%s` rather than string templates?

Comment (anxolin) [reply src/repos/dfusion/DfusionRepoImpl.ts]: Yes!, no a super big one though. The log is not necessarily enabled, so if you interpolate and then call the fn, then you interpolate for nothing :) 

Anyways, as I said, I don't care that much about optimizing at that level, so sometimes I use string interpolation if I feel that the code is more readable, what I find more important.

Comment (alfetopito) [reply src/repos/dfusion/DfusionRepoImpl.ts]: Got it!

File: test/sandbox/contract/getPastDeposits.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ *  SANDBOX: Prints events topics
+ *  RUN:     yarn sandbox test/sandbox/contract/getPastDeposits.ts
+ */
+const log = new Logger('sandbox:contract/getPastDeposits')
+
+async function exec (): Promise<void> {
+  log.info('Get past deposits for contract: %s', stableCoinConverterContract.options.address)
+  const events = await stableCoinConverterContract.getPastEvents('Deposit', { fromBlock: 0, toBlock: 'latest' })
```
Comment (Velenir) [inline test/sandbox/contract/getPastDeposits.ts]: This thing will eventually run into `Error: Returned error: query returned more than 1000 results`
'member that one?

File: src/contracts/index.ts [hunk 3]
```diff
@@ -9,7 +9,10 @@ function getStableConverterContract(): StablecoinConverter {
   const stableCoinContractAddress = STABLE_COIN_CONTRACT_ADDRESS as string
 
   const abi = require('./StablecoinConverter.json')
-  return new web3.eth.Contract(abi, stableCoinContractAddress) as StablecoinConverter
+
+  // FIXME: There's an issue with this conversion
```
Comment (Velenir) [inline src/contracts/index.ts]: No idea why it requires `unknown`. Not necessary in dex-react

File: test/sandbox/contract/getPastDeposits.ts (hunk 2 above)
Comment (anxolin) [reply test/sandbox/contract/getPastDeposits.ts]: Yep, that's just a sandbox. If that happens, we'll add a starting block or sth :)

File: src/contracts/index.ts (hunk 3 above)
Comment (anxolin) [reply src/contracts/index.ts]: I showed to nick, also knowledgeable in TS and struggled to understand why

Thanks for trying anyways

**Pull request merged**